src/apps/routes/schemas/containers.clj

```diff
@@ -210,17 +210,101 @@
    {:container_volumes_from [VolumesFrom]}
    "The list of VolumeFroms associated with a tool's container."))
 
-(def DevicesParamOptional     (s/optional-key :container_devices))
-(def VolumesParamOptional     (s/optional-key :container_volumes))
-(def VolumesFromParamOptional (s/optional-key :container_volumes_from))
+(s/defschema Port
+  (describe
+   {:id                              s/Uuid
+    (s/optional-key :host_port)      Integer
+    :container_port                  Integer
+    (s/optional-key :bind_to_host)   Boolean}
+   "Port information for a tool container."))
+
+(s/defschema NewPort
+  (describe
+   (dissoc Port :id)
+   "A map for adding a new port configuration to a tool container."))
+
+(s/defschema HostPort
+  (describe
+   {:host_port (s/maybe Integer)}
+   "The port on the host system from which requests are forwarded."))
+
+(s/defschema ContainerPort
+  (describe
+   {:container_port Integer}
+   "The port that the container is listening on. Info is forwarded to it from the host."))
+
+(s/defschema BindToHost
+  (describe
+   {:bind_to_host (s/maybe Boolean)}
+   "Whether or not to forward info from the host port to the container port."))
+
+(s/defschema ProxySettings
+  (describe
+   {:id                             s/Uuid
+    :image                          String
+    :name                           String
+    (s/optional-key :frontend_url)  String
+    (s/optional-key :cas_url)       String
+    (s/optional-key :cas_validate)  String
+    (s/optional-key :ssl_cert_path) String
+    (s/optional-key :ssl_key_path)  String}
+   "Interactive app settings for the reverse proxy that runs on the HTCondor nodes for each job."))
+
+(s/defschema NewProxySettings
+  (describe
+   (dissoc ProxySettings :id)
+   "A map for adding new interactive app reverse proxy settings."))
+
+(s/defschema ProxySettingsImage
```

Comment:
I'm only finding refereces to `ProxySettings` and `NewProxySettings`.
Can we remove these other `ProxySettings*` schemas?